libpod/container_internal_linux.go

```diff
@@ -369,13 +369,46 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		if err != nil {
 			return nil, err
 		}
-		volMount := spec.Mount{
-			Type:        "bind",
-			Source:      mountPoint,
-			Destination: namedVol.Dest,
-			Options:     namedVol.Options,
+
+		overlayFlag := false
+		for _, o := range namedVol.Options {
+			if o == "O" {
+				overlayFlag = true
+			}
+		}
+
+		if overlayFlag {
+			contentDir, err := overlay.TempDir(c.config.StaticDir, c.RootUID(), c.RootGID())
+			if err != nil {
+				return nil, err
+			}
+			overlayMount, err := overlay.Mount(contentDir, mountPoint, namedVol.Dest, c.RootUID(), c.RootGID(), c.runtime.store.GraphOptions())
```

Comment:
Does anything need to be done to unmount the volune when we're done?

Reply:
Cleanup for all the overlays are handled generically from base i.e `c.config.StaticDir` so it is handled as regular cleanup when needed https://github.com/containers/podman/blob/main/libpod/container_internal_linux.go#L2590 i have verified for both `fuse-overlay` and `native` overlay everything gets cleaned up.

Reply:
yes.